# HG changeset patch
# User Rik <rik@octave.org>
# Date 1522100414 25200
#      Mon Mar 26 14:40:14 2018 -0700
# Branch stable
# Node ID db1e3f3664f4429fe4accc518c5794ff9d115504
# Parent  2365c2661b3c6fa784b7a53dabab34a41f2c161d
doc: Update documentation for SEEK_SET, SEEK_CUR, SEEK_END.

* io.txi: Add DOCSTRING entries for SEEK_CUR, SEEK_END.

* file-io.cc (Ffseek): Re-order presentation of offsets to be SEEK_SET,
SEEK_CUR, SEEK_END.  Add SEEK_SET, SEEK_CUR, SEEK_END to @seealso list.
* file-io.cc (FSEEK_SET): Document only SEEK_SET, not SEEK_CUR and SEEK_END as
well.
* file-io.cc (FSEEK_CUR, FSEEK_END): Add @seealso links to other SEEK_XXX
functions and also to fseek.

diff --git a/doc/interpreter/io.txi b/doc/interpreter/io.txi
--- a/doc/interpreter/io.txi
+++ b/doc/interpreter/io.txi
@@ -1074,16 +1074,20 @@ Three functions are available for settin
 the file pointer for a given file.
 
 @DOCSTRING(ftell)
 
 @DOCSTRING(fseek)
 
 @DOCSTRING(SEEK_SET)
 
+@DOCSTRING(SEEK_CUR)
+
+@DOCSTRING(SEEK_END)
+
 @DOCSTRING(frewind)
 
 The following example stores the current file position in the variable
 @code{marker}, moves the pointer to the beginning of the file, reads
 four characters, and then returns to the original position.
 
 @example
 @group
diff --git a/libinterp/corefcn/file-io.cc b/libinterp/corefcn/file-io.cc
--- a/libinterp/corefcn/file-io.cc
+++ b/libinterp/corefcn/file-io.cc
@@ -716,26 +716,26 @@ is equivalent to @code{fseek (@var{fid},
 
 DEFMETHOD (fseek, interp, args, ,
            doc: /* -*- texinfo -*-
 @deftypefn  {} {} fseek (@var{fid}, @var{offset})
 @deftypefnx {} {} fseek (@var{fid}, @var{offset}, @var{origin})
 @deftypefnx {} {@var{status} =} fseek (@dots{})
 Set the file pointer to the location @var{offset} within the file @var{fid}.
 
-The pointer is positioned @var{offset} characters from the @var{origin},
-which may be one of the predefined variables @w{@code{SEEK_CUR}} (current
-position), @w{@code{SEEK_SET}} (beginning), or @w{@code{SEEK_END}} (end of
-file) or strings @nospell{@qcode{"cof"}}, @nospell{@qcode{"bof"}}, or
+The pointer is positioned @var{offset} characters from the @var{origin}, which
+may be one of the predefined variables @w{@code{SEEK_SET}} (beginning),
+@w{@code{SEEK_CUR}} (current position), or @w{@code{SEEK_END}} (end of file) or
+strings @nospell{@qcode{"bof"}}, @nospell{@qcode{"cof"}}, or
 @nospell{@qcode{"eof"}}.  If @var{origin} is omitted, @w{@code{SEEK_SET}} is
 assumed.  @var{offset} may be positive, negative, or zero but not all
 combinations of @var{origin} and @var{offset} can be realized.
 
 @code{fseek} returns 0 on success and -1 on error.
-@seealso{fskipl, frewind, ftell, fopen}
+@seealso{fskipl, frewind, ftell, fopen, SEEK_SET, SEEK_CUR, SEEK_END}
 @end deftypefn */)
 {
   int nargin = args.length ();
 
   if (nargin < 2 || nargin > 3)
     print_usage ();
 
   octave::stream_list& streams = interp.get_stream_list ();
@@ -3079,59 +3079,46 @@ environment variable.
 {
   if (args.length () != 0)
     print_usage ();
 
   return ovl (get_P_tmpdir ());
 }
 
 // NOTE: the values of SEEK_SET, SEEK_CUR, and SEEK_END have to be
-// this way for Matlab compatibility.
+//       this way for Matlab compatibility.
 
 DEFUNX ("SEEK_SET", FSEEK_SET, args, ,
         doc: /* -*- texinfo -*-
 @deftypefn  {} {} SEEK_SET ()
-@deftypefnx {} {} SEEK_CUR ()
-@deftypefnx {} {} SEEK_END ()
-Return the numerical value to pass to @code{fseek} to perform one of the
-following actions:
-
-@table @code
-@item SEEK_SET
-Position file relative to the beginning.
-
-@item SEEK_CUR
-Position file relative to the current position.
-
-@item SEEK_END
-Position file relative to the end.
-@end table
-@seealso{fseek}
+Return the numerical value to pass to @code{fseek} to position the file pointer
+relative to the beginning of the file.
+@seealso{SEEK_CUR, SEEK_END, fseek}
 @end deftypefn */)
 {
   return const_value ("SEEK_SET", args, -1);
 }
 
 DEFUNX ("SEEK_CUR", FSEEK_CUR, args, ,
         doc: /* -*- texinfo -*-
 @deftypefn {} {} SEEK_CUR ()
-Return the numerical value to pass to @code{fseek} to
-position the file pointer relative to the current position.
-@seealso{SEEK_SET, SEEK_END}
+Return the numerical value to pass to @code{fseek} to position the file pointer
+relative to the current position.
+@seealso{SEEK_SET, SEEK_END, fseek}
 @end deftypefn */)
 {
   return const_value ("SEEK_CUR", args, 0);
 }
 
 DEFUNX ("SEEK_END", FSEEK_END, args, ,
         doc: /* -*- texinfo -*-
 @deftypefn {} {} SEEK_END ()
-Return the numerical value to pass to @code{fseek} to
-position the file pointer relative to the end of the file.
-@seealso{SEEK_SET, SEEK_CUR}
+Return the numerical value to pass to @code{fseek} to position the file pointer
+relative to the end of the file.
+@seealso{SEEK_SET, SEEK_CUR, fseek}
 @end deftypefn */)
 {
   return const_value ("SEEK_END", args, 1);
 }
 
 static octave_value
 const_value (const char *, const octave_value_list& args,
              const octave_value& val)
